src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -2,18 +2,87 @@
 
 public class MyHashMap<K, V> implements MyMap<K, V> {
 
+    private static final int DEFAULT_SIZE = 16;
+    private static final double LOAD_FACTOR = 0.75;
+    private static final int MULTIPLIER = 2;
+    private int size;
+    private Node<K,V>[] table;
+
+    public MyHashMap() {
+        table = new Node[DEFAULT_SIZE];
+    }
+
     @Override
     public void put(K key, V value) {
-
+        if (size > table.length * LOAD_FACTOR) {
+            resize();
+        }
+        int index = getIndex(key);
+        Node<K, V> currentNode = table[index];
+        Node<K, V> newNode = new Node<>(key, value, null);
+        while (currentNode != null) {
+            if (currentNode.key == key
```

Comment:
Use 'Objects.equals()' for key comparison to handle both nulls and non-nulls without an explicit null check.